### PR TITLE
Update to Blender 2.9 and ID maps

### PR DESCRIPTION
**Pull request by mvoelk**
It seems that the script is still used by many as a starting point for their own work and it is often adapted to their needs.

For this reason I updated it to Blender 2.9 and added ID maps as an example for segmentation ground truth.

### CODE REVIEW

**Comment by panmari** [comment]
Thanks! I'll test it on my machine in the coming days, then merge.

One question I have is: What are ID maps? Are they a well known concept that doesn't need explaining?

**Comment by mvoelk** [comment]
@panmari I hope so :D ID maps, segmentation masks... the terminology varies in the literature.

I found a video by Adam Kelly that explains the concept very well, especially in the context of blender.
https://www.youtube.com/watch?v=xeprI8hJAH8

File: render_blender.py [hunk 1]
```diff
@@ -26,149 +30,209 @@
                     help='Number of bit per channel used for output. Either 8 or 16.')
 parser.add_argument('--format', type=str, default='PNG',
                     help='Format of files generated. Either PNG or OPEN_EXR')
+parser.add_argument('--resolution', type=int, default=600,
+                    help='Resolution of the images.')
```
Comment (panmari) [inline render_blender.py:34]: Point out that output is always square.

File: render_blender.py [hunk 2]
```diff
@@ -26,149 +30,209 @@
...
 argv = sys.argv[sys.argv.index("--") + 1:]
 args = parser.parse_args(argv)
 
-import bpy
+# Set up rendering
+context = bpy.context
+scene = bpy.context.scene
+render = bpy.context.scene.render
+
+render.engine = 'CYCLES' # ('CYCLES', 'BLENDER_EEVEE', ...)
```
Comment (panmari) [inline render_blender.py:44]: Why not leave the default here, i.e. delete the line?

File: render_blender.py [hunk 3]
```diff
@@ -26,149 +30,209 @@
...
+render.image_settings.color_mode = 'RGBA' # ('RGB', 'RGBA', ...)
+render.image_settings.color_depth = args.color_depth # ('8', '16')
+render.image_settings.file_format = args.format # ('PNG', 'OPEN_EXR', 'JPEG, ...)
+render.resolution_x = args.resolution
+render.resolution_y = args.resolution
+render.resolution_percentage = 100
```
Comment (panmari) [inline render_blender.py:50]: Necessary?

File: render_blender.py [hunk 4]
```diff
@@ -26,149 +30,209 @@
...
     bpy.ops.render.render(write_still=True)  # render still
 
-    b_empty.rotation_euler[2] += radians(stepsize)
+    # remove frame number from file names
+    for n in glob(render_file_path+'_*'):
+        os.rename(n, re.sub(r'(\d+)\.([^\.]+)$', r'.\2', n))
+
+    cam_empty.rotation_euler[2] += math.radians(stepsize)
+
+#bpy.ops.wm.save_as_mainfile(filepath='debug.blend')
```
Comment (panmari) [inline render_blender.py:238]: Remove

File: render_blender.py [hunk 5]
```diff
@@ -26,149 +30,209 @@
...
+    albedo_file_output.file_slots[0].path = render_file_path + "_albedo"
+    id_file_output.file_slots[0].path = render_file_path + "_id"
 
     bpy.ops.render.render(write_still=True)  # render still
 
-    b_empty.rotation_euler[2] += radians(stepsize)
+    # remove frame number from file names
+    for n in glob(render_file_path+'_*'):
+        os.rename(n, re.sub(r'(\d+)\.([^\.]+)$', r'.\2', n))
+
```
Comment (panmari) [inline render_blender.py:235]: The intent here is to remove the trailing 0001 from the files, right? Is this necessary? Might cause some confusion as blender reports 

> Saved: /tmp/teapot/teapot_r_324_depth0001.png

**Review by panmari — COMMENTED** [review]
Thanks again for the PR! Left a few comments.

File: render_blender.py (hunk 2 above)
Comment (mvoelk) [reply render_blender.py:44]: Some people may want to use Eevee

File: render_blender.py (hunk 3 above)
Comment (mvoelk) [reply render_blender.py:50]: Not sure what exactly you mean... Always square is better than fixed and always square. Smaller resolution renders faster when playing around with the code.

File: render_blender.py (hunk 4 above)
Comment (mvoelk) [reply render_blender.py:238]: As I mentioned, I think most of the users adjust the script to there needs. Users that have a background in engineering or whatever, but are often not that familiar with blender...

File: render_blender.py (hunk 5 above)
Comment (mvoelk) [reply render_blender.py:235]: Can you explain that in a little more detail? Is there an option in blender to turn it off? Is there a warning, if so what does it say? It make sense to keep the frame number when rendering sequences for movies and not doing the naming manually by the script...

I'll leave that to you ;)

File: render_blender.py (hunk 4 above)
Comment (panmari) [reply render_blender.py:238]: Ok, then comment what this line does.

File: render_blender.py (hunk 5 above)
Comment (panmari) [reply render_blender.py:235]: I quoted the standard output produced when calling the script. I'd rather be consistent with that output. In other words, please remove the renaming logic.

File: render_blender.py (hunk 2 above)
Comment (panmari) [reply render_blender.py:44]: I agree, the default from blender is even Eevee. I would like the script to stay consistent with that default.

Would your prefer having this exposed as a command line argument?

File: render_blender.py (hunk 3 above)
Comment (panmari) [reply render_blender.py:50]: I see, so you're saying that's an easy way (apart from the resolution argument), do adapt the size of the output. Correct?

Can you put that into a code comment here?

**Review by panmari — COMMENTED** [review]
Sorry for the long delay and thanks for your replies!

**Comment by mvoelk** [comment]
@panmari Sorry for being so lazy. Please take the code and do whatever you like... :)

File: render_blender.py (hunk 2 above)
Comment (TennisGazelle) [reply render_blender.py:44]: I think it's best to leave this here to help indicate to developers who do not have much experience with BPY (such as myself) to change certain hyper parameter, even if we're not going to make them cmd-line-argument configuratble.

Setting variables and their examples of their possible values, even if it's defaulted correctly, is very helpful for open sourcing.

**Comment by panmari** [comment]
Ok, I'll patch a few things after merging then.

**Pull request merged**